src/amount.h

```diff
@@ -128,6 +128,21 @@ struct CTokenAmount { // simple std::pair is less informative
         this->nValue -= amount;
         return Res::Ok();
     }
+    CAmount SubWithRemainder(CAmount amount) {
+        // safety checks
+        if (amount < 0) {
+            Add(-amount);
+            return 0;
```

Comment:
Why need to support substract negative amount, will it happen, can return error instead?

Reply:
As you can see `CAmount SubWithRemainder(CAmount amount);` returns CAmount type, not error type. So we need support negative amount substraction and return remainder anyway.